### DISCUSSION TITLE
Refi and Gcloud Billing ? Does it only Read/write the changed documents or everything?

### DISCUSSION THREAD

**Discussion by mshenawy22** [thanhlmm/refi-app#91] [Q&A]
Hi everyone,
I love Revi , Hats off for every developer who worked on this.
I have a question which may seem basic about how committing the changes work.
Obviously in some cases there are enormous amount of data that are being pulled from FireStore.
I was wondering how Revi commits the data , Does it only writes the new documents or the changed ones which is the best case scenario or Does it do any redundant reads and writes to documents and collections?

Appreciate the answer if anybody knows.

Thanks All.

**Comment by thanhlmm** [comment]
Refi only writes the new document or the changed one!

**Reply by mshenawy22** [reply]
What about reading operations ,Any information on How it handles reading  Collections and documents
Does it read all the collections once you connect to the database or only the ones you open?

**Reply by thanhlmm** [reply]
it only read the one that you open. You can check the source code to confirm
